dessia_common/datatools/dataset.py

```diff
@@ -666,11 +663,20 @@ def _histogram_unic_value(self, idx_col: int, name_attr: str):
     def _tooltip_attributes(self):
         return self.common_attributes
 
-    def _plot_data_list(self):
-        plot_data_list = []
-        for row, _ in enumerate(self.dessia_objects):
-            plot_data_list.append({attr: self.matrix[row][col] for col, attr in enumerate(self.common_attributes)})
-        return plot_data_list
+    def _object_to_sample(self, dessia_object: DessiaObject, reference_path: str, row: int):
+        sample_values = {attr: self.matrix[row][col] for col, attr in enumerate(self.common_attributes)}
+        full_reference_path = f"{reference_path}/dessia_objects/{row}"
```

Comment:
Don't be afraid to overwrite reference_path name, here. that's what I would do, instead of defining a new variable

Reply:
Actually, reference_path is a sequence and when you rewrite it, it behaves like an empty list:
`reference_path = f"{reference_path}/dessia_objects/{row}"` is taking previously created reference path to update it... so you get a "dessia_objects/0/dessia_objects/1/dessia_objects/2/..." reference path

Reply:
I think it is because the old version of your function (that did not return a sample). I am pretty sure the reference_path value did not behave like a sequence (well, it is, but a str, so its immutable), but, as you built it, returned it and reuse in the for loop in your `to_samples` method, it got updated every step in the loop. 

With the new way of defining your function, you should not have the same behavior, because reference_path's modification scope keeps being restricted to the `object_to_sample` method